esphome/components/bp5758d/bp5758d.cpp

```diff
@@ -39,7 +39,12 @@ void BP5758D::loop() {
   uint8_t data[17];
   if (this->pwm_amounts_[0] == 0 && this->pwm_amounts_[1] == 0 && this->pwm_amounts_[2] == 0 &&
       this->pwm_amounts_[3] == 0 && this->pwm_amounts_[4] == 0) {
-    // Off / Sleep
+    // First turn all channels off
+    data[0] = BP5758D_MODEL_ID + BP5758D_ADDR_START_3CH;
+    for (int i = 1; i < 16; i++)
+      data[i] = 0;
+    this->write_buffer_(data, 17);
+    // Then sleep
     data[0] = BP5758D_MODEL_ID + BP5758D_ADDR_STANDBY;
     for (int i = 1; i < 16; i++)
       data[i] = 0;
```

Comment:
No need to loop and fill the buffer twice.
```suggestion
    for (int i = 1; i < 16; i++)
      data[i] = 0;

    // First turn all channels off
    data[0] = BP5758D_MODEL_ID + BP5758D_ADDR_START_3CH;
    this->write_buffer_(data, 17);
    // Then sleep
    data[0] = BP5758D_MODEL_ID + BP5758D_ADDR_STANDBY;
```